File: 백준/Silver/2606. 바이러스/바이러스.cc
#include <iostream>
#include <vector>

using namespace std;

int dfs(int node, const vector<vector<int>>& graph, vector<bool>& visited) {
	visited[node] = true;
	int count = 0;

	for (int idx = 1; idx <= graph.size(); idx++) {
		if (!visited[idx] && graph[node][idx] == 1)
			count += (1 + dfs(idx, graph, visited));
	}

	return count;
}

int main() {
	int n, m;

	cin >> n >> m;

	vector<vector<int>> graph(n + 1, vector<int>(n + 1, 0));
	vector<bool> visited(n, false);

	for (int idx = 0; idx < m; idx++) {
		int a, b;
		
		cin >> a >> b;

		graph[a][b] = 1;
		graph[b][a] = 1;
	}

	int result = dfs(1, graph, visited);

	cout << result << endl;

	return 0;
}

File: 백준/Silver/2606. 바이러스/바이러스.py
import sys
sys.setrecursionlimit(10000)

def dfs(v, graph, visited):
    visited[v] = True
    for nxt in graph[v]:
        if not visited[nxt]:
            dfs(nxt, graph, visited)


def main():
    input = sys.stdin.readline
    n = int(input().strip())           # 컴퓨터 수
    m = int(input().strip())           # 연결 쌍 수
    graph = {i: [] for i in range(1, n+1)}
    for _ in range(m):
        a, b = map(int, input().split())
        graph[a].append(b)
        graph[b].append(a)

    # 방문 순서를 위한 정렬(필수는 아님)
    for k in graph:
        graph[k].sort()

    visited = [False] * (n+1)
    dfs(1, graph, visited)

    # 1번 컴퓨터를 제외한 감염된 컴퓨터 수
    infected = sum(1 for i in range(2, n+1) if visited[i])
    print(infected)

if __name__ == '__main__':
    main()
